pkg/cleaner/cleaner.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleaner
+
+import (
+	"fmt"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"time"
+)
+
+type Cleaner struct {
+	config *rest.Config
+}
+
+// NewCleaner returns new Cleaner struct
+func NewCleaner() (*Cleaner, error) {
+	k8sKubeconfig, err := rest.InClusterConfig()
+	if err != nil {
+		return &Cleaner{}, fmt.Errorf("failed to get Kubernetes client config: %v", err)
+	}
+
+	return &Cleaner{k8sKubeconfig}, nil
+}
+
+// RemoveCRDs takes three steps, first scale down controlle manager pod, second removes all finalizers from
+// CRs and the last step removes all CRDs with specific label
+func (c *Cleaner) RemoveCRDs(releaseName, releaseNamespace, controllerManagerName string) error {
```

Comment:
Discussed f2f:

1. Remove the CRD (for each execute Delete action)
2. Check if CRs still exits if yes then remove the Service Catalog finalizers 
2.1. Wait till all finalizers are removed
3. Check if all CRDs from Service Catalog were removed


benefit from removing the CRDs before CRs:
- User will not be able to add new CRs
  ```
  Error from server (MethodNotAllowed): error when creating "broker.yaml": the server does not allow this method on the requested resource (post clusterservicebrokers.servicecatalog.k8s.io)
  ```